services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowCrudSpec.groovy

```diff
@@ -560,4 +499,104 @@ class FlowCrudSpec extends BaseSpecification {
     Switch findSwitch(SwitchId swId) {
         topologyDefinition.activeSwitches.find { it.dpId == swId }
     }
+
+    def getConflictingData() {
+        [
+            [
+                conflict: "the same vlans on the same port on src switch",
+                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
+                    flowToConflict.source.portNumber = dominantFlow.source.portNumber
+                    flowToConflict.source.vlanId = dominantFlow.source.vlanId
+                },
+                getError: { FlowPayload flowToError, String operation = "create" ->
+                    portError(operation, flowToError.source.portNumber, flowToError.source.datapath, flowToError.id)
+                }
+            ],
+            [
+                conflict: "the same vlans on the same port on dst switch",
+                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
+                    flowToConflict.destination.portNumber = dominantFlow.destination.portNumber
+                    flowToConflict.destination.vlanId = dominantFlow.destination.vlanId
+                },
+                getError: { FlowPayload flowToError, String operation = "create" ->
+                    portError(operation, flowToError.destination.portNumber, flowToError.destination.datapath,
+                            flowToError.id)
+                }
+            ],
+            [
+                conflict: "no vlan vs vlan on the same port on src switch",
+                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
+                    flowToConflict.source.portNumber = dominantFlow.source.portNumber
+                    flowToConflict.source.vlanId = 0
+                },
+                getError: { FlowPayload flowToError, String operation = "create" ->
+                    portError(operation, flowToError.source.portNumber, flowToError.source.datapath, flowToError.id)
+                }
+            ],
+            [
+                conflict: "no vlan vs vlan on the same port on dst switch",
+                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
+                    flowToConflict.destination.portNumber = dominantFlow.destination.portNumber
+                    flowToConflict.destination.vlanId = 0
+                },
+                getError: { FlowPayload flowToError, String operation = "create" ->
+                    portError(operation, flowToError.destination.portNumber, flowToError.destination.datapath,
+                            flowToError.id)
+                }
+            ],
+            [
+                conflict: "vlan vs no vlan on the same port on src switch",
```

Comment:
Given that a top-level flow basically consists of forward and reverse flows, do we really need both "vlan vs no vlan" and "no vlan vs vlan" test cases?

Reply:
Maybe not but these tests are very fast and I would prefer to keep them. They don't almost take any time.

Reply:
I added those cases in panic when there was a prod defect with vlans to exclude the possibility of any missing scenarios.